_posts/2021-01-11-GPS-serverless.md

```diff
@@ -0,0 +1,195 @@
+
+It is an essential step where allows me to send data between The Thing Network and AWS.
+
+Here is the sample code for lambda written in typescript
+
+```js
+export const saveItemToDynamoDb = async (
+  tableName: string,
+  dynamoDBAttributes: unknown,
+  dynamoDB: DynamoDB
+) => {
+  const attributes: PutItemInputAttributeMap = DynamoDB.Converter.marshall(
+    dynamoDBAttributes
+  );
+
+  const parameter: PutItemInput = {
+    TableName: tableName,
+    Item: attributes,
+  };
+
+  try {
+    const savedItem = await dynamoDB.putItem(parameter).promise();
+    const response = DynamoDB.Converter.unmarshall(savedItem.Attributes);
+    return response;
+  } catch (error) {
+    console.log("There is an error on putting object into table", error);
+  }
+};
+```
+
+The HTTP endpoint will invoke this lambda. Hence, any data from The Thing Network (TTN) will get send into this lambda.
+Those data will be stored in dynamoDB that I created in the previous step.
+
+Once I have the HTTP endpoint, I am ready to integrate AWS and The Thing Network (TTN).
+
+## Fifth Step: Integration between The Thing Network (TTN) and AWS
+
+Integration method available in The Thing Network (TTN):
+
+<center><img src="/img/blog/serverless-GPS/integrationMethod.png" /></center><br/>
+
+In order to have it integrate with the http endpoint that I created in previous step, I will go with HTTP integration method here.
+
+## Sixth Step: Create the application - React App
+
+**Before I start**
+
+- You will need to know how to [Create a React App](https://create-react-app.dev/docs/getting-started/)
+
+  To check the data quickly, I have created a simple React application with a simple map to show the tracking device's coordinates.
+
+## Seventh Step: Connect the database with React App via AWS Appsync
+
+Now I have almost everything ready, and I will need the React application to pull data from the dynamoDB via API.
+
+I have chosen [AWS Appsync](https://aws.amazon.com/appsync/). AWS Appsync is a fully managed service that makes it easy to develop GraphQL API. I have a straightforward schema as the application is relatively simple.
+
+**Appsync Schema**
+
+```yaml
+Type: AWS::AppSync::GraphQLSchema
+Properties:
+  ApiId: !GetAtt ServerlessTrackerAppsyncApi.ApiId
+  Definition: |
+    type Uplink {
+      dev_id: String
+      payload_fields: Payload
+    }
+    type Payload {
+      Latitude: Float
+      Longitude: Float
+      ALARM_status: String
+    }
+    type Query {
+      getUplinkData(dev_id: String, time: String): [Uplink]
+    }
+    schema {
+      query: Query
+    }
+```
+
+## Wrapping Up
+
+In conclusion, those are the seven steps that I used to build a simple GPS tracking device.
+
+I hope this blog post has inspired you to think about how you can apply serverless technology to your problem.
+
+Thanks for reading. If you would like to know more about serverlesss, [contact with us](https://mechanicalrock.io/lets-get-started).
```

Comment:
Great post Sher. May I suggest here summing up the seven steps, re-iterating what you've just gone over, maybe in bullet point form. It may also be worth saying how you can now track your dog more easily and how it's actually saved you time or whatever. i.e. real-world benefit.